NAS-117953 / 22.12 / NAS-117953: Add 'serializeConfig' method

bvasilenko opened the pull request:
Fix for a bug when user clicks **Edit** and array fields are empty.

For testing, please see example in the ticket

bugclerk commented:
Jira URL: https://ixsystems.atlassian.net/browse/NAS-117953

RehanY147 commented on src/app/services/app-schema.service.ts:
```diff
@@ -615,4 +615,27 @@ export class AppSchemaService {
+            }
+          });
+        } else if (_.isPlainObject(valueConfig)) {
+          this.serializeConfig(valueConfig, formConfig);
```
Not sure what this method is meant to do. Code needs to be more readable.

bvasilenko commented:
@RehanY147 

> Shouldn't change the array while we're traversing it

Good one. Now it returns a modified value, and there's no mutable reference arguments

> Not sure what this method is meant to do. Code needs to be more readable.

Now I've renamed the method and added javadoc-style commentary

RehanY147 commented on src/app/services/app-schema.service.ts:
```diff
@@ -615,4 +615,45 @@ export class AppSchemaService {
+  /**
+   * Restores keys from form group
+   * @param config Object without keys. Example: { objectList: [{ nestedList: ['test4', 'test5'] }] }
+   * @param form Form group to restore keys from
```
form Form group

RehanY147 commented on src/app/services/app-schema.service.ts:
```diff
@@ -615,4 +615,45 @@ export class AppSchemaService {
+   * Restores keys from form group
+   * @param config Object without keys. Example: { objectList: [{ nestedList: ['test4', 'test5'] }] }
+   * @param form Form group to restore keys from
+   * @returns Object with keys. Example: { objectList: [{ nestedList: ['test4', 'test5'] }] }
```
The example is the same as `@param config`

RehanY147 reviewed (approved):
Looks good. Nice!
